compliance_checker/ioos.py

```diff
@@ -812,24 +837,40 @@ def check_single_platform(self, ds):
             val = False
             results.append(Result(BaseCheck.HIGH, val, "platform", [msg]))
 
-
         elif ((not glb_platform) and num_platforms > 0):
             msg = "If platform variables exist, a global attribute \"platform\" must also exist"
             val = False
             results.append(Result(BaseCheck.HIGH, val, "platform", [msg]))
 
         elif num_platforms == 0 and glb_platform:
-            msg = "A dataset with a global \"platform\" attribute must platform have variables"
+            msg = "A dataset with a global \"platform\" attribute must have platform variables"
             val = False
             results.append(Result(BaseCheck.HIGH, val, "platform", [msg]))
 
         elif num_platforms == 0 and (not glb_platform):
+            # can't determine if the data should actually have a platform or not,
+            # so this must result in a pass
             msg = "Gridded model datasets are not required to declare a platform"
```

Comment:
Each case only has one message though. If there is only one message and then the return, I see no reason to create the list outside of the return and append to it. It just seems like it would add more lines:

```
msgs = []
if ...:
    msgs.append("...")
    results.append(Result(BaseCheck...))
elif ...:
    msgs.append("...")
    results.append("...")
...
``` 
and so on. Thoughts?

Reply:
Ah, was confused as to what you meant here. You meant "Only add the message if it's as error". While this behavior occurs in multiple places, no need for it to continue. I'll implement this.